kratos/utilities/data_type_traits.h

```diff
@@ -231,6 +252,27 @@ template<class TDataType, std::size_t TSize> class DataTypeTraits<array_1d<TData
     ///@name Public static operations
     ///@{
 
+    /**
+     * @brief Returns whther the given @ref TCheckIndex is dynamic.
+     *
+     * This method returns true if the @ref TCheckIndex of the dimension
+     * corresponds to a dynamic type. If not, this returns false.
+     *
+     * @tparam TCheckIndex          User input dimension index.
+     * @tparam TCurrentIndex        Used for the recursion only. Do not specify this.
+     * @return true                 If the input dimension index corresponds to dynamic data type.
+     * @return false                If the input dimension index corresponds to static data type.
+     */
+    template<unsigned int TCheckIndex, unsigned int TCurrentIndex = 0>
+    static constexpr inline bool IsDimensionDynamic()
+    {
+        if constexpr(TCheckIndex == TCurrentIndex) {
+            return false;
+        } else {
+            return ValueTrait::template IsDimensionDynamic<TCheckIndex, TCurrentIndex + 1>();
```

Comment:
```suggestion
            return ValueTraits::template IsDimensionDynamic<TCheckIndex, TCurrentIndex + 1>();
```

Reply:
btw this will shoot off to an infinite loop **at compile time** if someone messes up the template params. Plz add a `constexpr` branch to prevent such cases.

Reply:
This will be used in `HDFFile` where i do branching for MPI communication based on the dynamic types onle since doing MPI communication for static dimensions are redundant andcostly.

Reply:
As far as I see, it cannot go recursively in an infinite loop since, you cannot define a data type without having a primitive type at the end.

Reply:
BTW, changed to `ValueTraits` ;)